### DISCUSSION TITLE
Can't read keys. Invalid input.

### DISCUSSION THREAD

**Discussion by edwinps** [krzyzanowskim/ObjectivePGP#202] [General]
- ObjectivePGP version:  0.15.0
- Platform (iOS/macOS):  iOS 13.7
- Installation method (CocoaPods/Carthage/...): CocoaPods
- Project name:

# Description

When trying to read the attached public key (create with Android), I get the following error

 *** Assertion failure in +[PGPPacketFactory packetWithData:offset:consumedBytes:], /Users/marcinkrzyzanowski/Devel/ObjectivePGP/ObjectivePGP/PGPPacketFactory.m:130
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Object type not satisfying: '(null)' must be of type 'PGPPacket' but is '(null)'.'
*** First throw call stack:
(0x1bf161654 0x1bee83bcc 0x1bf0646ec 0x1bf4aa16c 0x109b2a7ec 0x109b2f16c 0x109b2e514 0x1011acfdc 0x1011af80c 0x100ee8bb0 0x100eae3c4 0x100c0dc20 0x1c283d990 0x1c283e2bc 0x1c2af1be0 0x1c2af62a0 0x1c2bf94e4 0x1c36bd2e4 0x1c36916e8 0x1c3691bf0 0x1c3691d40 0x1c5c7ed7c 0x10b353730 0x10b361710 0x1bf0df6bc 0x1bf0da590 0x1bf0d9ba8 0x1c9250344 0x1c32153e4 0x100d3db5c 0x1bef618f0)
libc++abi.dylib: terminating with uncaught exception of type NSException

Importing this key works with android and web, but with iOS there is a crash and doesn't have the key

# Steps to reproduce
let dataPublicKey = Data(base64Encoded: publicKeyString)!
try? ObjectivePGP.readKeys(from: dataPublicKey)

string ->
"mQENBF98eukBCACeLKwfdWGBkMjGFIn6d7b4wxbXsTea5lDUvY5FRvBTV3ZGSIuRGO+z9TatwD/zANgGNkK/xrFOvH/0qaTk8RcwQpVIrtbzHg1OjC98LUBSf7Yql6LhqVdqRgFq/fCL5Xz8THs26dUa4Sw07HORgLWcmC+12+CFByHSgm/4OqcM4gBv+9zVCKjCD7REjsHuXFjVM14YsIDHSLYQphBxiuBXtQqtc5S+ZdxIJ3ai4lCe3fPUMFK0GtEjVHrFTs31KOng2ujx8Y/hO5PHhE+4vXbO7Cuf1IXXVh0dDq9lRo9Mhjlz7OCfVGdDcCr2PUXMDHgno/OUAhmCTit/c0SLlvBbABEBAAGJAZwEEAMCAAYFAl98euwACgkQULipqwqHybMP6gv+LWCvusSw7XLTJaiZ684a5uxpZreCEUPuHA46xU8HJLoh6Q4OykcHDv5bi6YEadBI6GD+zi5gWLbM6exT7BQ52oJ1qw9/BKwjPc+84N8652PuUywKrt1FS3SjopDfnSOFF1RVA+L1S6/+3gneVou/20TfNUvlRv+1PrsKyC2OeDkv+4eHs40TXw1P6yUMZ27deIngMwyX7OwC1owcuCGXvbkfldNEV6snbmEzvZIIdQffY3ane38grKz2JATjguisgmOcXBaC+nuqj1sdAKM8T9AI4uzzXhvLbtVCd0E692+Www4xThugQmUVmh8bmW/jDmVx9HV1+wsQtnLqyuq531VxXrMeZLCNfsKqlUUWnSlwJze2xo6lfMiK3DNxtyz+Tu7a5cE0GUfSrb0lUQo6BZWE6tRrJyn57LnuySLvdTFcZUYw/qa67jj9xAyIVgR/OI/HUJeCEr6gfur3UUIY4yecmCvMDFeGyL//6dLq8nz1doQQbKMnQNuHQxb27bRUtD9BbmQgMDE3IDwxYzI2Mzc3Zjk2ODQ4Y2UxYWQ1OTJhMDJiZjIzYTcwNTk1ZWJhYzJmQGRldi5heW9iYS5tZT6JASIEEwEIAAwFAl98eukCGyECGQEACgkQauyYoc3pPFmxPQgAhn+ydKipYAOqLVo7Nv5Q7awv7mK2AeQjKF210pseiocw1CiS72YBi2lM0cSbLAZH8RQYo99L2v/KVYRz6GbRynTlGGXOaZxMjScGiSEtyK5Dwav3WdvPMjYSAifh5Dzh/3GRgtVqBpELTsWKug4VXYqPUD/zs6fZtcPJSsWgasIBXqFXySPd9X6zTW9hRVg99ZStpSOb1UREj4GBnu1F7397cegRLIjIiI1cVfXoIawSD7l6nVyPHIZXJA1KLs2TgArSOOGPmsSGbwrwPGxPnEa0GmNWcax6F8x1vLBxi+7Ny6+lMWswjp9OFAp+5fwCMfA8MuSzsKEQ5dTudSzB+7hSBF98eukTCCqGSM49AwEHAgMEsohncMVoSPYy8aq5rxljBrZLLwrxh0mhZdGTwKO2PFXmvTHbfOuUoAHIRTaWA/pSlq2y46lKPVt2u0/Ay/AkTIkBfwQYAQgAaQUCX3x66QIbAl8gBBkTCAAGBQJffHrpAAoJEANt+2GWdgShQw0BAOisUL4fryAgXkVCEeZgvv3caaSu5ozuQaSn37cftSDuAQDSEBAlw/i+jhsyuJzLhwUI2tOTaUttKcdDEMo5YrrC+AAKCRBq7Jihzek8WT2uB/sG7470SXmnW+ad+E9OxpFXL4+0x1H0OOUbK5C9tfqaAPVl7Z9YrvrzFciexTxDtq/QXRdaf86hb2t+0fGXoMUfdyWUztWuXjieyzgPm/VWy5d52WTSr/6RfeAMxL5vOvJ8Sa4RJMVLgYEy/LciCSlAHzA7SM5BXc4t9a9j3egwMDhNYCeI1+DKMp8aEKcJt3sW1m5a62eZ7wgqypswy0UbZ0809hddXh6JuP2A1TcC8PrzfWpdmpZnoYT2ebJ4PlN3mqi+9SytYyq9QbGXEgGOG2ABnWqghqmofKczs6vUW+ifRUTK/X0BLb4VfMS0V7U6SqBWsdsUY+wzLSjyC+YJuFYEX3x66RIIKoZIzj0DAQcCAwQ6Fbz7Bal/vUapDeSb4k/pBUY4XInQ2LXAuIlujCmMNU2xUSojtKjytaWs3iHQ+w3VxfhkH4vI0bxtYVtw9ys2AwEIB4kBHwQYAQgACQUCX3x66QIbBAAKCRBq7Jihzek8WZ2mB/9/VYdlaPfmzNh4Q/IxqNj22eXGbXx2Jr96MeMpFF0wjGKsTR++9cUnSeEU9/tAYVbP3aBdM/C+sCeEuqZOy8SvT5lIyMqflxetaBrgbVwV66AhVD2d20QD/9J9u7ItFiGlfJfw+5GfRp15ItxdfZB9cSYju4LN7h7IApFIrJquPGs7dAV2XZJ04rXQ5mG9xPwrXgT82TgQk1BhpGX2g/l7tXejlni4oThjYc3498b8CA2zCrpD5zPtVZndgN3IVvqvxNuD7/B+D7Pgmz3vXXpZVa+S5O7rozkO4nX0rSvOiOxw+EOcztcdIqP+IpaQ2H36ncc9T9KN5YWoWSsh9OKU"

**Comment by vindicatesociety** [comment]
Any updates for this? here is the Open PGP code I want to rewrite in Swift5. I don't see any guidance on the below.
Thanks!! 😊 


import { createMessage, encrypt, readKey } from 'openpgp'

// Object to be encrypted
interface CardDetails {
 number?: string,    // required when storing card details
 cvv?: string        // required when cardVerification is set to cvv
}

// Encrypted result
interface EncryptedValue {
 encryptedData: string,
 keyId: string
}
 
const pciEncryptionKey = await getPCIPublicKey()
 
/**
* Encrypt card data function
*/
return async function(dataToEncrypt: CardDetails): Promise<EncryptedValue> {
 const decodedPublicKey = await readKey({ armoredKey: atob(publicKey) })
  const message = await createMessage({ text: JSON.stringify(dataToEncrypt) })
  return encrypt({
    message,
    encryptionKeys: decodedPublicKey,
  }).then((ciphertext) => {
    return {
      encryptedMessage: btoa(ciphertext),
      keyId,
    }
  })
}

**Comment by vindicatesociety** [comment]
var pgpkey = try! ObjectivePGP.readKeys(from:Data(base64Encoded: key)!) 
let encrypted = try! ObjectivePGP.encrypt(try! JSONSerialization.data(withJSONObject: cardInfo), addSignature: false, using: pgpkey)

😏 in my face. this works!!
